golang/go.mod

```diff
@@ -1,15 +1,11 @@
 module example.com/app
 
-go 1.22
+go 1.17
```

Comment:
The primary runtime target for these bootstraps are the devcontainers, as we can easily control the go version and other system dependencies there. Anyone with an outdated version of go can just launch the devcontainer inside VSCode or a Codespace/Gitpod (or update their system which they should be doing anyways. Running an outdated version should never be the system default imho).

The reasoning is maintenance: Ideally, renovate will keep the bootstraps updated to the latest available version including all dependencies, including major versions etc. Only if an update fails a PR will be raised that requires my attention. 

If we set the go version to `1.17`, will that limit the range of available updates of our dependencies? E.g. if `testify@v2` will only support `go@1.22`, will that keep renovatebot from updating testify? Putting it differently, I would not downgrade the minimum version here if it potentially implies a maximum version of a dependency in the future. The maxim must be that we'll never even start lagging behind versions, otherwise the bootstraps will start rotting.

Does that make sense? :sweat_smile: